common/src/main/scala/com/gu/sfl/persistence/SavedArticlesPersistence.scala

```diff
@@ -55,21 +53,20 @@ class SavedArticlesPersistenceImpl(persistanceConfig: PersistenceConfig) extends
     }
   }
 
-  override def write(userId: String, savedArticles: SavedArticles): Try[Option[SavedArticles]] = {
-    scanamo.exec(table.put(DynamoSavedArticles(userId, savedArticles))) match {
-      case Some(Right(articles)) =>
-        logger.debug(s"Succcesfully saved articles for $userId")
-        Success(Some(articles.ordered))
-      case Some(Left(error)) =>
-        val exception = new IllegalArgumentException(s"$error")
-        logger.debug(s"Exception Thrown saving articles for $userId:", exception)
-        Failure(exception)
-      case None => {
-        logger.debug(s"Successfully saved but none retrieved for $userId")
-        Success(Some(savedArticles))
-      }
-    }
-  }
+  /** *
+   * Using the update method as a create or update method, because
+   * dynamo update allows us to return newly created values:
+   * https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_UpdateItem.html
+   * Previously we had been using the put method to create new records,
+   * however the put operation only supports return values of the old value or none
+   * https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_PutItem.html
+   * Since we use the database response to respond to
```

Comment:
This explanation tails off a little here!

Reply:
Thanks good spot